flex/paths.py

```diff
@@ -156,7 +156,8 @@ def match_path_to_api_path(path_definitions, target_path, base_path='', global_p
         matches = [(p, r.match(target_path)) for p, r in paths.items()]
```

Comment:
Instead of using this line, which is in effect a duplication of [line 147](https://github.com/pipermerriam/flex/pull/59/files#diff-79b7810358d612d76f53cf572cbc0f20R147), I think it'd be better to remove this line, and replace line 147 with the following.

```python
matching_api_paths = [(p, r.match(target_path)) for p, r in paths.items() if r.match(target_path)]
```
Now that `matching_api_paths` will be a list of 2-tuples of (*api_path*, *path_regex_match*), all of the places where `matching_api_paths` is accessed will need to be changed accordingly.

- [line 166](https://github.com/pipermerriam/flex/pull/59/files#diff-79b7810358d612d76f53cf572cbc0f20R166) will need to change to something like `[v[0] for v in matching_api_paths]` in order to get at the inner API paths and not dump a bunch of regex matches into the error message.
- [line 170](https://github.com/pipermerriam/flex/pull/59/files#diff-79b7810358d612d76f53cf572cbc0f20R170) will need to change to `return matching_api_paths[0][0]` to get at the api_path in the inner tuple.


Once those are cleaned up, we can remove the `matches` variable and just use `matching_api_paths` in its place.

Does that make sense?